tee-worker/core/rest-client/src/rest_client.rs

```diff
@@ -182,6 +185,36 @@ where
 	}
 }
 
+#[cfg(feature = "async")]
+#[async_trait]
+impl<H> RestGet for RestClient<H>
+where
+	H: SendHttpRequest + std::marker::Send,
+{
+	/// Make a GET request.
+	async fn get<U, T>(&mut self, params: U) -> Result<T, Error>
+	where
+		T: serde::de::DeserializeOwned + RestPath<U> + std::marker::Send,
+		U: std::marker::Send,
+	{
+		let body = self.make_request::<U, T>(Method::GET, params, None, None)?;
+
+		serde_json::from_str(body.as_str()).map_err(|err| Error::DeserializeParseError(err, body))
+	}
```

Comment:
If response is != 200, will it throw a `DeserializeParseError` error? could that be misleading? 

furthermore, should we keep this generic and not assume the body's content is json string?